core/string/ustring.cpp

```diff
@@ -4770,8 +4835,13 @@ String String::sprintf(const Array &values, bool *error) const {
 						return "not enough arguments for format string";
 					}
 
-					if (!values[value_index].is_num()) {
-						return "* wants number";
+					Variant::Type value_type = values[value_index].get_type();
+					if (!values[value_index].is_num()
+							&& value_type != Variant::VECTOR2
+							&& value_type != Variant::VECTOR3
+							&& value_type != Variant::VECTOR4 && value_type != Variant::VECTOR2I
+							&& value_type != Variant::VECTOR3I && value_type != Variant::VECTOR4I) {
```

Comment:
That looks inconsistent.

I suggested:
```
&& 2 && 2I
&& 3 && 3I
&& 4 && 4I
```
It can also be this if you prefer, though it's less common in the codebase to separate float and int variants in such groups:
```
&& 2 && 3 && 4
&& 2I && 3I && 4I
```

Reply:
Fixed. I was a bit too quick there. Sorry